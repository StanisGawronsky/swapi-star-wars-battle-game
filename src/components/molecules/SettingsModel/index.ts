export { default as SettingsModel } from "./SettingsModel";
