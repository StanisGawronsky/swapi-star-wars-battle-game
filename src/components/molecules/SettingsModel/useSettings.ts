import { useEffect, useState } from "react";
interface ISettings {
  currentAttribute: string;
  currentPlayerType: string;
  playerTypes: string[];
  availableAttributes: string[];
  setCurrentPlayerType: (player: string) => void;
  setCurrentAttribute: (attribute: string) => void;
}
const CHAMPIONS = "champions";
const STARSHIPS = "starships";
const CHAMPION_ATTRIBUTES: string[] = ["height", "mass", "birth_year"];
const STARSHIP_ATTRIBUTES: string[] = [
  "cost_in_credits",
  "length",
  "max_atmosphering_speed",
  "crew",
  "passengers",
  "cargo_capacity",
  "consumables",
  "hyperdrive_rating",
  "MGLT",
];
export const useSettings = (): ISettings => {
  const [currentAttribute, setCurrentAttribute] = useState<string>(
    CHAMPION_ATTRIBUTES[0]
  );
  const [availableAttributes, setAvailableAttributes] =
    useState<Array<string>>(CHAMPION_ATTRIBUTES);
  const [currentPlayerType, setCurrentPlayerType] = useState(CHAMPIONS[0]);
  const playerTypes = [CHAMPIONS, STARSHIPS];

  useEffect(() => {
    if (currentPlayerType === CHAMPIONS) {
      setAvailableAttributes(CHAMPION_ATTRIBUTES);
      return setCurrentAttribute(CHAMPION_ATTRIBUTES[0]);
    }

    if (currentPlayerType === STARSHIPS) {
      setAvailableAttributes(STARSHIP_ATTRIBUTES);
      return setCurrentAttribute(STARSHIP_ATTRIBUTES[0]);
    }
  }, [currentPlayerType]);

  return {
    currentAttribute,
    setCurrentAttribute,
    currentPlayerType,
    setCurrentPlayerType,
    playerTypes,
    availableAttributes,
  };
};
