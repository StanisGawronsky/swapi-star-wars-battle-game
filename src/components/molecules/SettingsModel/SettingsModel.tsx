import React from "react";
import { Settings } from "../Settings";
import { useSettings } from "./useSettings";

const SettingsModel = () => {
  const props = useSettings();
  return <Settings {...props} />;
};

export default SettingsModel;
