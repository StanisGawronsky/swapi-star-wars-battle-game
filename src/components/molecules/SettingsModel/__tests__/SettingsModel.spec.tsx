import React from "react";
import { render } from "@testing-library/react";
import SettingsModel from "../SettingsModel";
describe("<SettingsModel />", () => {
  it("renders correctly", () => {
    const { container } = render(<SettingsModel />);
    expect(container).toMatchSnapshot();
  });
});
