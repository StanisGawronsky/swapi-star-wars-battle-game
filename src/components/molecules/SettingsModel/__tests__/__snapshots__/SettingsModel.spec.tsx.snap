// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsModel /> renders correctly 1`] = `
<div>
  <div>
    <div>
      <label>
        Chose your player type
      </label>
      <select>
        <option
          value="champions"
        >
          champions
        </option>
        <option
          value="starships"
        >
          starships
        </option>
      </select>
    </div>
    <div>
      <label>
        Chose your attribute
      </label>
      <select>
        <option
          value="height"
        >
          height
        </option>
        <option
          value="mass"
        >
          mass
        </option>
        <option
          value="birth_year"
        >
          birth year
        </option>
      </select>
    </div>
  </div>
</div>
`;
